catalog/core/catalog-core-directorymonitor/src/main/java/org/codice/ddf/catalog/content/monitor/DavAlterationObserver.java

```diff
@@ -0,0 +1,318 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.content.monitor;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.io.UncheckedIOException;
+import java.util.Comparator;
+import java.util.LinkedHashSet;
+import java.util.List;
+
+import com.github.sardine.DavResource;
+import com.github.sardine.Sardine;
+
+/**
+ * A webdav implementation of the {@link org.apache.commons.io.monitor.FileAlterationObserver}
+ * Uses https://github.com/lookfirst/sardine
+ */
+public class DavAlterationObserver implements Serializable {
+
+    private static final long serialVersionUID = 42L;
+
+    private static final DavResource[] EMPTY_RESOURCES = new DavResource[] {};
+
+    private transient LinkedHashSet<EntryAlterationListener> listeners = new LinkedHashSet<>();
+
+    private final DavEntry rootEntry;
```

Comment:
✏️ Nit. You might want to move the `rootEntry` down so the non-serialized and serialized fields in the class are all grouped together.